apps/aecore/test/aec_conductor_tests.erl

```diff
@@ -235,6 +236,39 @@ test_preemption() ->
     %% TODO: check the transaction pool
     ok.
 
+test_preemption_pulled() ->
+    %% Assert preconditions
+    assert_stopped_and_genesis_at_top(),
+
+    %% Generate a chain
+    Chain = aec_test_utils:gen_blocks_only_chain(7),
+    {Chain1, Chain2} = lists:split(3, Chain),
+    Top1 = lists:last(Chain1),
+    Top2 = lists:last(Chain2),
+    Hash1 = block_hash(Top1),
+    Hash2 = block_hash(Top2),
+
+    %% Seed the server with the first part of the chain
+    [ok = ?TEST_MODULE:add_synced_block(B) || B <- Chain1],
+    wait_for_top_block_hash(Hash1),
+
+    %% Start mining and make sure we are starting
+    %% from the correct hash.
+    true = aec_events:subscribe(start_mining),
+    aec_conductor:start_mining(),
+
+    wait_for_start_mining(Hash1),
+
+    %% Sync the rest of the chain, which will take over.
+    [?TEST_MODULE:add_synced_block(B) || B <- Chain2],
+    wait_for_top_block_hash(Hash2),
+
+    %% The mining should now have been preempted
+    %% and started over with the new top block hash
+    wait_for_start_mining(Hash2),
```

Comment:
Cherry-picking this commit on master (aa2d03e8) - i.e. code without fix in this PR - I got error as expected:
```
    aec_conductor_tests: chain_test_ (Test preemption of mining by block pulled from a network peer)...*failed*
in function aec_conductor_tests:wait_for_gproc/2 (/Users/luca/dev/aeternity/epoch/_build/test/lib/aecore/test/aec_conductor_tests.erl, line 474)
in call from aec_conductor_tests:wait_for_start_mining/1 (/Users/luca/dev/aeternity/epoch/_build/test/lib/aecore/test/aec_conductor_tests.erl, line 465)
in call from aec_conductor_tests:test_preemption_pulled/0 (/Users/luca/dev/aeternity/epoch/_build/test/lib/aecore/test/aec_conductor_tests.erl, line 268)
**error:{timeout,block_created}
  output:<<"">>
```